aspnet/migration/http-modules/sample/Asp.Net5/src/Asp.Net5/Middleware/HttpContextDemoMiddleware.cs

```diff
@@ -0,0 +1,320 @@
+            // even though they are used to create the response.
+
+            // ----------------------
+            // ASP.NET 4 - HttpContext.Response.Headers  
+            // ASP.NET 4 - HttpContext.Response.Cookies  
+            // 
+            // Issue with all response headers is that they will not be sent after anything has been written to the
+            // response body.
+            // To solve this, use OnStarting to set a callback that will be called right before headers are sent.
+            // Set the headers in that callback.
+            //
+            // Setting a cookie results in a Set-Cookie response header.
+            // So use an OnStarting handler here as well.
+
+            //>40
+            httpContext.Response.OnStarting(SetHeaders, state: httpContext);
+            httpContext.Response.OnStarting(SetCookies, state: httpContext);
+            //<40
+
+            // ==================================================================
+            // Context
+
+            // Unique request ID (no ASP.NET 4 counterpart)
+            //>01
+            string requestId = httpContext.TraceIdentifier;
+            //<01
+
+            // ASP.NET 4 - HttpContext.Items
+            //>01b
+            IDictionary<object, object> items = httpContext.Items;
+            //<01b
+
+            // ==================================================================
+            // Request
+
+            // ASP.NET 4 - HttpContext.Request.HttpMethod
+            //>02
+            string httpMethod = httpContext.Request.Method;
+            //<02
+
+            // ----------
+
+            // ASP.NET 4 - HttpContext.Request.QueryString
+            //>03
+            IReadableStringCollection queryParameters = httpContext.Request.Query;
+
+            // If no query parameter "key" used, values will have 0 items
+            // If single value used for a key (...?key=v1), values will have 1 item ("v1")
+            // If key has multiple values (...?key=v1&key=v2), values will have 2 items ("v1" and "v2")
+            IList<string> values = queryParameters["key"];
+
+            // If no query parameter "key" used, value will be ""
+            // If single value used for a key (...?key=v1), value will be "v1"
+            // If key has multiple values (...?key=v1&key=v2), value will be "v1,v2"
+            string value = queryParameters["key"].ToString();
+            //<03
+
+            // ----------
+
+            // ASP.NET 4 - HttpContext.Request.Url and HttpContext.Request.RawUrl
+            //>04
+            string scheme = httpContext.Request.Scheme;
+            string host = httpContext.Request.Host.Value;
+            string pathBase = httpContext.Request.PathBase.Value;
+            string path = httpContext.Request.Path.Value;
+            string queryString = httpContext.Request.QueryString.Value;
+            string url = scheme + "://" + host + pathBase + path + queryString;
```

Comment:
No, this will often give you the wrong result when generating urls. Use GetDisplayUrl or GetEncodedUrl (https://github.com/aspnet/HttpAbstractions/blob/dev/src/Microsoft.AspNetCore.Http.Extensions/UriHelper.cs).